src/classes/SObjectRepository.cls

```diff
@@ -2,40 +2,102 @@
 * This file is part of the Nebula Framework project, released under the MIT License.             *
 * See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
 *************************************************************************************************/
-public abstract class SObjectRepository extends NebulaCore implements ISObjectRepository {
+public virtual class SObjectRepository extends NebulaCore implements ISObjectRepository, ICrud {
 
+    private final Schema.SObjectType sObjectType;
+    private final Schema.FieldSet fieldSet;
+    private final List<Schema.SObjectField> sobjectFieldList;
+    private final Schema.SObjectField idField;
     protected IQueryBuilder queryBuilder;
 
-    protected SObjectRepository(Schema.SObjectType sobjectType, Schema.FieldSet fieldSet) {
+    public SObjectRepository(Schema.SObjectType sobjectType, Schema.FieldSet fieldSet) {
         this(sobjectType, fieldSet, null);
     }
 
-    protected SObjectRepository(Schema.SObjectType sobjectType, List<Schema.SObjectField> sobjectFieldList) {
+    public SObjectRepository(Schema.SObjectType sobjectType, List<Schema.SObjectField> sobjectFieldList) {
         this(sobjectType, null, sobjectFieldList);
     }
 
     private SObjectRepository(Schema.SObjectType sobjectType, Schema.FieldSet fieldSet, List<Schema.SObjectField> sobjectFieldList) {
         this.currentModule = NebulaCore.Module.REPOSITORY;
 
-        this.queryBuilder = new QueryBuilder(sobjectType, fieldSet, sobjectFieldList);
+        this.sObjectType = sObjectType;
+        this.fieldSet = fieldSet;
+        this.sObjectFieldList = sObjectFieldList;
+        this.idField = this.getField(this.sObjectType, 'Id');
+    }
+
+    protected IQueryBuilder queryBuilder {
+        get {
+            return new QueryBuilder(sObjectType, fieldSet, sObjectFieldList);
+        }
+    }
+
+    public IQueryBuilder getQueryBuilder() {
+        return this.queryBuilder;
+    }
+    //SOQL
+    public virtual SObject getById(Id recordId) {
+        return this.queryBuilder
+            .whereField(new List<QueryPredicate>{QueryPredicate.EQUALS(this.idField, recordId)})
```

Comment:
Related to my confusion on the benefits of QueryPredicate, line 42 seems more complex than the original way. This seems easier to ready to me:
`whereField(this.idField, QueryOperator.EQUALS, recordId)`